File: PyQuest/src/ql/ast/expressions/variable_node.py
from ql.ast.expressions.expression_node import ExpressionNode


class VariableNode(ExpressionNode):
    def __init__(self, metadata, expression_type, identifier, value):
        super(VariableNode, self).__init__(metadata, expression_type, value)
        self.__identifier = identifier

    @property
    def identifier(self):
        return self.__identifier

File: ForcePush/src/main/java/org/uva/forcepushql/parser/ast/elements/expressionnodes/VariableNode.java
package org.uva.forcepushql.parser.ast.elements.expressionnodes;

import org.uva.forcepushql.parser.ast.visitors.ASTExpressionVisitor;
import org.uva.forcepushql.parser.ast.visitors.ASTVisitor;
import org.uva.forcepushql.parser.ast.elements.ExpressionNode;

public class VariableNode extends ExpressionNode
{
    public String name;

    public VariableNode() {
        super(true);
    }

    public String getName()
    {
        return name;
    }

    public void setName(String name)
    {
        this.name = name;
    }

    @Override
    public String accept(ASTVisitor visitor)
    {
        return visitor.visit(this);
    }

    @Override
    public Boolean accept(ASTExpressionVisitor visitor)
    {
        return visitor.visit(this);
    }
}
